python/cudf/cudf/core/dataframe.py

```diff
@@ -3079,6 +3079,8 @@ def insert(self, loc, name, value):
             )
 
         if _is_scalar_or_zero_d_array(value):
+            if value is cudf.NA:
+                value = None
```

Comment:
This works but I think it would be slightly better to enable handling of this case inside `scalar_broadcast_to`. Specifically [these lines](https://github.com/rapidsai/cudf/blob/branch-21.10/python/cudf/cudf/utils/utils.py#L45-L48) look at lot like [this](https://github.com/rapidsai/cudf/blob/branch-21.10/python/cudf/cudf/_lib/scalar.pyx#L547-L553)

Reply:
We may want to handle this in scalar_broadcast_to

Reply:
@brandon-b-miller @beckernick ok, I moved the changes to `scalar_broadcast_to`, thanks!

Reply:
Functionally this LGTM but I also think we should just use the existing function `cudf._lib.scalar._is_null_host_scalar` since I think it is exactly the same logic

Reply:
I see what you mean - updated